packages/component-library/src/components/plmg-text-input/test/plmg-text-input.spec.tsx

```diff
@@ -0,0 +1,145 @@
+    <div class="plmg-text-input-wrapper">
+    <label class="small plmg-text-input-label" htmlfor="small">
+    small
+    </label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+        <input class="small" id="small" type="text" name="small">
+        </div>
+    </plmg-text-input>
+    `);
+  });
+  it('renders labeless', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input show-label="false" label="Hidden"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input label="Hidden" show-label="false">
+    <div class="plmg-text-input-wrapper">
+    <label class="medium plmg-text-input-label visually-hidden" htmlfor="hidden">
+    Hidden
+    </label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+      <input class="medium" id="hidden" type="text" name="Hidden">
+        </div>
+    </plmg-text-input>
+    `);
+  });
+  it('renders tip', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input tip="Tip Text" label="Tip"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input label="Tip" tip="Tip Text">
+    <div class="plmg-text-input-wrapper">
+    <label class="medium plmg-text-input-label" htmlfor="tip">
+    Tip
+    </label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+      <input class="medium" id="tip" type="text" name="Tip">
+        </div>
+      <span class="medium plmg-text-input-tip">Tip Text</span>
+    </plmg-text-input>
+    `);
+  });
+  it('renders error', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input error-message="error" label="Error"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input label="Error" error-message="error">
+    <div class="plmg-text-input-wrapper">
+    <label class="medium plmg-text-input-label" htmlfor="error">Error</label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+      <input class="error medium" name="Error" id="error" type="text">
+        </div>
+        <plmg-error-message message="error" size="medium" style="margin-top: 8px;"></plmg-error-message>
+    `);
+  });
+  it('renders default value', async () => {
+    const page = await newSpecPage({
+      components: [TextInput],
+      html: `<plmg-text-input default-value="default value" label="Default Value"></plmg-text-input>`,
+    });
+    expect(page.root).toEqualHtml(`
+
+    <plmg-text-input default-value="default value" label="Default Value">
+    <div class="plmg-text-input-wrapper">
+    <label class="medium plmg-text-input-label" htmlfor="default-value">Default Value</label>
+      <div class="plmg-text-input-field-wrapper" tabindex="0">
+        <input class="medium" name="Default Value" id="default-value" type="text">
+      </div>
+    `);
+  });
```

Comment:
You removed default value as a prop. This test can be removed.
Furthermore, since we're using a protected form element, what is put into the input field is not represented on the DOM. As such a HTML comparison will not adequately test this use case.